src/overlays/actors/ovl_Obj_Aqua/z_obj_aqua.c

```diff
@@ -26,55 +35,266 @@ const ActorInit Obj_Aqua_InitVars = {
+void func_80ACBA10(ObjAqua* this) {
+    s32 pad;
+    MtxF sp2C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBA10.s")
+    func_800C0094(this->actor.floorPoly, this->actor.world.pos.x, this->actor.floorHeight, this->actor.world.pos.z,
+                  &sp2C);
+    func_8018219C(&sp2C, &this->actor.shape.rot, 0);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBA60.s")
+s32 func_80ACBA60(ObjAqua* this, GlobalContext* globalCtx) {
+    s32 pad;
+    WaterBox* waterBox;
+    f32 ySurface;
+    s32 bgId;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/ObjAqua_Init.s")
+    if (func_800C9EBC(globalCtx, &globalCtx->colCtx, this->actor.world.pos.x, this->actor.world.pos.z, &ySurface,
+                      &waterBox, &bgId) &&
+        (this->actor.world.pos.y < ySurface)) {
+        return 1;
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/ObjAqua_Destroy.s")
+void ObjAqua_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjAqua* this = THIS;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBC70.s")
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.scale.x = 0.0009f;
+    this->actor.scale.y = 0.0005f;
+    this->actor.scale.z = 0.0009f;
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 60.0f);
+    if (0) {};
+    this->actor.shape.shadowAlpha = 140;
+    this->alpha = 255;
+    if (func_80ACBA60(this, globalCtx)) {
+        for (i = 0; i < 8; i++) {
+            EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, -4.0f, 4.0f, 4.0f,
+                                 (Rand_ZeroOne() * 0.09f) + 0.03f);
+        }
+        func_80ACBDCC(this);
+    } else {
+        func_80ACBC70(this);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBC8C.s")
+void ObjAqua_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjAqua* this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBD34.s")
+void func_80ACBC70(ObjAqua* this) {
+    this->counter = 200;
+    this->actionFunc = func_80ACBC8C;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Aqua/func_80ACBD48.s")
+void func_80ACBC8C(ObjAqua* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 0x21) {
+        if (this->actor.bgCheckFlags & 1) {
+            func_80ACB7F4(this, globalCtx);
+            func_80ACBA10(this);
+            Audio_PlayActorSound2(&this->actor, 0x2920);
+            func_80ACBD34(this);
+        } else {
+            func_80ACB6A0(this, globalCtx);
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x28, 0x2817);
```

Comment:
```suggestion
            Audio_PlayActorSound2(&this->actor, NA_SE_EV_BOTTLE_WATERING);
            func_80ACBD34(this);
        } else {
            func_80ACB6A0(this, globalCtx);
            Audio_PlaySoundAtPosition(globalCtx, &this->actor.world.pos, 0x28, NA_SE_EV_BOMB_DROP_WATER);
```
The macros for these can be found in `include/sfx.h`!